base/applications/osk/settings.c

```diff
@@ -0,0 +1,102 @@
+﻿/*
+ * PROJECT:         ReactOS On-Screen Keyboard
+ * LICENSE:         GPL - See COPYING in the top level directory
+ * PURPOSE:         Settings file for warning dialog on startup
+ * PROGRAMMERS:     Copyright 2018 Bișoc George (fraizeraust99 at gmail dot com)
+ */
+
+/* INCLUDES *******************************************************************/
+
+#include "settings.h"
+
+/* DECLARATIONS ****************************************************************/
+
+BOOL bShowWarning = TRUE;
+
+/* FUNCTIONS *******************************************************************/
+
+BOOL LoadDataFromRegistry()
+{
+    HKEY hKey;
+    LONG lResult;
+    DWORD dwData;
+    DWORD cbData = sizeof(DWORD);
+
+    /* Open the key, so that we can query it */
+    lResult = RegOpenKeyEx(HKEY_CURRENT_USER,
+                           L"Software\\Microsoft\\osk",
+                           0,
+                           KEY_READ,
+                           &hKey);
+
+    if (lResult != ERROR_SUCCESS)
+    {
+        /* Bail out and return FALSE if we fail */
+        return FALSE;
+    }
+
+    /* Query the key */
+    lResult = RegQueryValueEx(hKey,
+                              L"ShowWarning",
+                              0,
+                              0,
+                              (BYTE *)&dwData,
+                              &cbData);
+
+    if (lResult != ERROR_SUCCESS)
+    {
+        /* Bail out and return FALSE if we fail */
+        return FALSE;
+    }
+
+    /* Load the data value (it can be either FALSE or TRUE depending on the data itself) */
+    bShowWarning = (dwData == 0 ? FALSE : TRUE);
```

Comment:
bShowWarning = (dwData != 0);

Reply:
`dwData == 0` is actually correct. Otherwise the dialog box will be displayed only once and the next attempt of launching the application the said box won't be displayed. The user has to launch the application again in order for the dialog welcome box being displayed again.

This can be seen the same on Magnify code:
https://github.com/reactos/reactos/blob/master/base/applications/magnify/settings.c#L62